src/DebuggerForm.cpp

```diff
@@ -588,10 +573,26 @@ void DebuggerForm::createForm()
 	dw->setClosable(true);
 	connect(dw, SIGNAL(visibilityChanged(DockableWidget*)),
 	        this, SLOT(dockWidgetVisibilityChanged(DockableWidget*)));
-	connect(this, SIGNAL(connected()),
-	        slotView, SLOT(refresh()));
-	connect(this, SIGNAL(breakStateEntered()),
-	        slotView, SLOT(refresh()));
+	connect(this, &DebuggerForm::connected, slotView, &SlotViewer::refresh);
+	connect(this, &DebuggerForm::breakStateEntered, slotView, &SlotViewer::refresh);
+
+	// create breakpoints viewer
+	bpView = new BreakpointViewer(this);
+	connect(bpView, &BreakpointViewer::contentsUpdated, this, &DebuggerForm::reloadBreakpoints);
+	connect(this, &DebuggerForm::breakpointsUpdated, bpView, &BreakpointViewer::sync);
+	dw = new DockableWidget(dockMan);
+	dw->setWidget(bpView);
+	dw->setTitle(tr("Debug list"));
+	dw->setId("DEBUG");
+	dw->setFloating(false);
+	dw->setDestroyable(false);
+	dw->setMovable(true);
+	dw->setClosable(true);
+	connect(dw, &DockableWidget::visibilityChanged, this,
+		&DebuggerForm::dockWidgetVisibilityChanged);
+	connect(this, &DebuggerForm::runStateEntered, bpView, &BreakpointViewer::setRunState);
+	connect(this, &DebuggerForm::breakStateEntered, bpView, &BreakpointViewer::setBreakState);
+	connect(slotView, &SlotViewer::slotsUpdated, bpView, &BreakpointViewer::sync); // TODO: change to a new function that only repaint bp colors
```

Comment:
In the previous patch I reviewed I suggested to replace a signal/slot connection by a direct call.
The title of this commit is "replace direct calls by signal/slots", so I was a bit suspicious. Is this a good idea?

But if I read this patch correctly, then all(?) signal/slot connections are between different components, right? In that case I'm more neutral about the change.

Reply:
Replacing signals/slots by direct calls would force us to store references between BreakpointViewer, SlotViewer and DisasmViewer since slot changes in SlotViewer and toggle breakpoint in DisasmView both have effects on BreakpointViewer.